pkg/engine/journal_snapshot.go

```diff
@@ -465,6 +465,26 @@ func (ssm *sameSnapshotMutation) mustWrite(step deploy.Step) bool {
 		}
 	}
 
+	if len(old.PropertyDependencies) != len(new.PropertyDependencies) {
+		logging.V(9).Infof("SnapshotManager: mustWrite() true because of PropertyDependencies")
+		return true
+	}
+
+	if len(old.PropertyDependencies) > 0 || len(new.PropertyDependencies) > 0 {
```

Comment:
We don't need this check. From L468 we know that they have the same length, so `len(old.PropertyDependencies) > 0` implies `len(new.PropertyDependencies) > 0`. The `for` loop does all the checking we need here.